src/python/WMCore/BossAir/Plugins/SimpleCondorPlugin.py

```diff
@@ -0,0 +1,590 @@
+                logging.error("Ignoring this, but this is very strange")
+            else:
+                logging.error("No job report for job with id %s and gridid %s", job['id'], job['gridid'])
+
+                if os.path.isfile(reportName):
+                    os.remove(reportName)
+
+                # create a report from scratch
+                condorReport = Report()
+                logOutput = 'Could not find jobReport\n'
+
+                if os.path.isdir(job['cache_dir']):
+                    # TODO: Alan - how about we look at the .out or .err file instead of the .log ?
+                    condorLog = os.path.join(job['cache_dir'], "condor.%s.log" % job['gridid'])
+                    if os.path.isfile(condorLog):
+                        logTail = BasicAlgos.tail(condorLog, 50)
+                        logOutput += 'Adding end of condor.log to error message:\n'
+                        logOutput += '\n'.join(logTail)
+                    condorReport.addError("NoJobReport", 99303, "NoJobReport", logOutput)
+                else:
+                    msg = "Serious Error in Completing condor job with id %s!\n" % job['id']
+                    msg += "Could not find jobCache directory %s\n" % job['cache_dir']
+                    msg += "Creating a new cache_dir for failed job report\n"
+                    logging.error(msg)
+                    os.makedirs(job['cache_dir'])
+                    condorReport.addError("NoJobReport", 99304, "NoCacheDir", logOutput)
+
+                condorReport.save(filename=reportName)
+
+                logging.debug("Created failed job report for job with id %s and gridid %s", job['id'], job['gridid'])
+
+        return
+
+    def updateSiteInformation(self, jobs, siteName, excludeSite):
+        """
+        _updateSiteInformation_
+
+        Allow or disallow jobs to run at a site.
+        Called externally by Ops scripts if a site enters or leaves Down, Draining or Aborted.
+
+        Kill job if after removing site from allowed sites it has nowhere to run.
+
+        Parameters:    excludeSite = False when moving to Normal
+                       excludeSite = True when moving to Down, Draining or Aborted
+        """
+        schedd = htcondor.Schedd()
+
+        jobtokill = []
+        try:
+            itobj = schedd.xquery("WMAgent_AgentName == %s" % classad.quote(self.agent),
+                                  ['ClusterId', 'ProcId', 'JobStatus', 'DESIRED_Sites', 'ExtDESIRED_Sites'])
+        except:
+            return jobtokill
+        else:
+            jobInfo = {}
+            for jobAd in itobj:
+                gridId = "%s.%s" % (jobAd['ClusterId'], jobAd['ProcId'])
+                jobInfo[gridId] = jobAd
+            for job in jobs:
+                jobAd = jobInfo.get(job['gridid'], None)
+                if jobAd:
+                    # only deal with Idle jobs
+                    if str(jobAd.get('JobStatus', 100)) == '1':
```

Comment:
How about we remove this line and instead add a constraint like `JobStatus == 1` or so to the `xquery` command?